kcalcore/recurrencerule.cpp

```diff
@@ -623,6 +623,16 @@ void Constraint::appendDateTime( const QDate &date, const QTime &time,
                                  QList<KDateTime> &list ) const
 {
   KDateTime dt( date, time, timespec );
+  KDateTime nonTimespecDt( date, time, KDateTime::Spec::UTC() );
+  if ( !dt.isValid() && nonTimespecDt.isValid() && timespec.utcOffset() == 0 ) {
+    // This is to fix an issue where CalDAV servers can provide a VTIMEZONE
+    // definition for CEST (eg "Europe/Berlin") with a DAYLIGHT component like:
+    // "DTSTART:19700329T020000\r\n"
+    // "RRULE:FREQ=YEARLY;BYDAY=-1SU;BYMONTH=3\r\n"
+    // This will fail after 1981 since the system definition differs, resulting
+    // in an invalid KDateTime due to the "invalid" timespec.
```

Comment:
I didn't print out the spectype tbh, but I suspect that it was Invalid, yes.  All I know is that the KDateTime becomes invalid if you use that member variable timespec, for dates AFTER 1981, while using the VTIMEZONE definition provided in testThree() (and also testOne() I assume).

But interestingly, in my tests, the utcOffset of the member variable timespec is ALWAYS zero!  The "time" variable passed to the function is already adjusted for the UTC offset defined by the VTIMEZONE definition, by that point.

In short, I suspect (but am not 100% certain) that we could _always_ use the system UTC timespec when creating the dt here.  But, I thought it safer to only "fall back" to that behaviour, obviously.